lib/processEventDataHelper.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+const moment = require('moment-timezone');
+const _ = require('lodash');
+const Q = require('q');
+
+
+
+function processEventData(cfg, messageBody) {
+
+  console.log("Processing event data ...");
+  var defer = Q.defer();
+
+  function checkRequiredFields(cfg, messageBody) {
+
+    console.log("Checking required fields ...");
+    if (!messageBody.start) {
+      return defer.reject(new Error("Start Time missing! This field is required!"));
+    }
+    if (!messageBody.end) {
+      return defer.reject(new Error("End Time missing! This field is required!"));
+    }
+    if (!cfg.timeZone) {
+      return defer.reject(new Error("Time Zone missing! This field is required!"));
+    }
+    if (!cfg.calendarId) {
+      return defer.reject(new Error("Calendar ID missing! This field is required!"));
+    }
+    if ((cfg.bodyContentType) && (!messageBody.body)) {
+      return defer.reject(new Error("Body Type provided, but Body Content is missing!"));
+    }
+
+    return buildPostBody(cfg, messageBody);
+  }
+
+  function buildPostBody() {
+
+    function formatDate(date, timeZone, format) {
+      if (moment(date).isValid()) {
+        return moment.tz(date, timeZone).format(format);
+      }
+      if (moment(date, 'x').isValid()) {
+        return moment(date, 'x').tz(timeZone).format(format);
+      }
+      return defer.reject(new Error("Invalid date " + date));
```

Comment:
Same questions here: This is a sync operation. Maybe the function should not be aware of any promises?